test/ecto/query/planner_test.exs

```diff
@@ -824,11 +824,22 @@ defmodule Ecto.Query.PlannerTest do
       from(Post, join: c in subquery(child)) |> normalize()
     end
 
-    assert_raise Ecto.QueryError, ~r/`parent_as\(:posts\)` can only be used in subqueries/, fn ->
+    assert_raise Ecto.QueryError, ~r/could not find named binding `parent_as\(:posts\)`/, fn ->
       from(Post, where: parent_as(:posts).code == ^123) |> normalize()
     end
   end
 
+  test "normalize: nested parent_as" do
+    child3 = from(c in Comment, where: parent_as(:posts).deleted == false, select: c.id)
+    child2 = from(c in Comment, where: c.id in subquery(child3), select: c.id)
+    child = from(c in Comment, where: parent_as(:posts).posted == c.posted and c.id in subquery(child2))
+
+    query = from(Post, as: :posts, join: c in subquery(child)) |> normalize()
+    assert Macro.to_string(hd(hd(query.joins).source.query.wheres).expr) =~ "parent_as(&0).posted() == &0.posted()"
+    assert Macro.to_string(hd(hd(query.joins).source.query.wheres).expr) =~ "in %Ecto.SubQuery{"
+  end
+
```

Comment:
This is not a good test. How can I improve it?

Reply:
It looks OK to me. The rest of the work needs to be done in ecto_sql to make it recursively.

Reply:
I wrote a simple integration test and it seems to be working out of the box (on postgres). Didn't commit it because I'm not sure about where it belongs.

Reply:
Hrm, good call. Sounds good to me then.

Reply:
I think the best is to write them as unit tests in ecto_sql. :)